app/controllers/application_controller.rb

```diff
@@ -210,6 +210,12 @@ def render_component(component_class, *args, **kwargs)
     render component_class.new(*args, **kwargs), content_type: "text/html"
   end
 
+  def creator_onboarding_mode
+    Settings::General.waiting_on_first_user
+    # || (current_user&.creator? && !current_user.saw_onboarding?)
```

Comment:
What path are we sending them down?

The "we can't use `current_user`" scenario only applies to actions with `set_cache_control_headers` set. This is why dashboard/settings/etc. have different rules.

Instead of adding this logic to all the views via `application.html.erb`, could we send them to a dedicated path which uses an alternate layout file here or something along those lines?

It seems like this could basically use the same logic that sends a normal user through onboarding, if we're also re-using `saw_onboarding`... But when we get to onboarding we could to display either the regular use path.

My question is... If we're in a path where we can't show `current_user`, i.e. `/`, the home page, then the problem will go deeper than just hiding/showing header footer, as we also can't check for `current_user` in the action.

I think if the logic first sends to a safe place to implement the onboarding functionality the way we would for any user but then applies different logic _there_? On that note, the `onboardings_controller` also applies caching via `set_cache_control_headers` but that isn't so critical and could likely be removed if it's the most convenient.

Reply:
This makes sense, thank you for the help! I looked into it further and I think I know how to go about implementing this now that you've outlined the details above. 

I'll pull down @juliannatetreault's branch once she's complete with it. It seems like we'd apply similar logic like what we've done for the `onboardings_controller` to the newly created `creator_settings_controller`. 

In the meantime, I've [created a PR for removing the header and footer for just the signup and email page.](https://github.com/forem/forem/pull/14816)